### PR TITLE
Listed all the memory relays from the 100 pickups

### PR DESCRIPTION
**Pull request by UltiNaruto**
Added pickup memory relay logging to resource_tracing.rs
Implemented id generation for custom pickups

### CODE REVIEW

File: src/bin/resource_tracing.rs [hunk 1]
```diff
@@ -525,6 +526,19 @@ fn extract_pickup_location<'r>(
...
     };
 
+    // Required for map dots
+    let memory_relay = search_for_scly_object(&obj.connections, &scly_db,
+        |obj| obj.property_data.as_memory_relay()
+            .map(|hm| hm.name.to_str().unwrap() == "Memory Relay")
+            .unwrap_or(false),
+    );
+
+    let memory_relay_id = if memory_relay.is_some() {
```
Comment (toasterparty) [inline src/bin/resource_tracing.rs:538]: Is this ever false? If not, please write this in a way that fails if something unexpected happens.

File: src/patches.rs [hunk 2]
```diff
@@ -12845,16 +12821,36 @@ fn build_and_run_patches(gc_disc: &mut structs::GcDisc, config: &PatchConfig, ve
...
+                                        instance_id: 0xffffffff,
+                                    },
+                                    hudmemo: pickup_meta::ScriptObjectLocation {
+                                        layer: 0,
+                                        instance_id: 0xffffffff,
+                                    },
+                                    memory_relay: pickup_meta::ScriptObjectLocation {
+                                        layer: 0,
+                                        instance_id: ((room_idx as u32) >> 16) | (0xffff - (custom_pickup_idx as u32)),
+                                    },
```
Comment (toasterparty) [inline src/patches.rs]: Why not just have `memory_relay: pickup_meta::ScriptObjectLocation` as your input instead of fabricating data?

**Review by toasterparty — CHANGES_REQUESTED** [review]
Please update comments like this one to reflect what is actually being done:

https://github.com/toasterparty/randomprime/pull/55/files#diff-4e76dd88afada4ac87706bb06b3bdc157c736afcb0e50db60232f31b0378c57aL1343-L1345

File: src/patches.rs (hunk 2 above)
Comment (UltiNaruto) [reply src/patches.rs]: separate the id and position in the args would be the way then

**Review by toasterparty — APPROVED** [review]

**Pull request merged**